Spigot-Server-Patches/0357-Add-source-block-to-BlockPhysicsEvent.patch

```diff
@@ -0,0 +1,22 @@
+From d81b977104ba95086af880aff9f93669ea99537f Mon Sep 17 00:00:00 2001
+From: Sotr <i@omc.hk>
+Date: Thu, 23 Aug 2018 16:14:12 +0800
+Subject: [PATCH] Add source block to BlockPhysicsEvent
+
+
+diff --git a/src/main/java/net/minecraft/server/World.java b/src/main/java/net/minecraft/server/World.java
+index 04d0fa1d..64d75934 100644
+--- a/src/main/java/net/minecraft/server/World.java
++++ b/src/main/java/net/minecraft/server/World.java
+@@ -590,7 +590,7 @@ public abstract class World implements IBlockAccess {
+                 // CraftBukkit start
+                 CraftWorld world = ((WorldServer) this).getWorld();
+                 if (world != null && !((WorldServer)this).stopPhysicsEvent) { // Paper
+-                    BlockPhysicsEvent event = new BlockPhysicsEvent(world.getBlockAt(blockposition.getX(), blockposition.getY(), blockposition.getZ()), CraftMagicNumbers.getId(block));
```

Comment:
could blockposition1 be null?

Reply:
checked the caller of `world.a`, i don't think this can be null